src/test/scala/com/eclipsesource/schema/SchemaValidatorSpec.scala

```diff
@@ -165,39 +167,37 @@ class SchemaValidatorSpec extends PlaySpecification {
       }
     }
 
-  "should resolve references on the classpath via UrlResolver" in {
-    val validator = SchemaValidator().addUrlResolver(ClasspathUrlResolver())
+  "should resolve references on the classpath with ClasspathUrlResolver" in {
+    val validator = SchemaValidator().addUrlResolver(ClasspathUrlProtocolHandler())
     // some.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve references on the classpath via UrlResolver and custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
-    // some.json references location.json within JAR
+      .addRelativeUrlHandler(new ClasspathHandler)
+    // some-issue-65.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some-issue-65.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve relative references with custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65 - 2)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
+      .addRelativeUrlHandler(new ClasspathHandler)
     val mySchemaJson = getClass.getResourceAsStream("/my-schema.schema")
     val mySchema = JsonSource.schemaFromStream(mySchemaJson)
     val result = validator.validate(mySchema.get, Json.obj("foo" -> Json.obj("bar" -> "whatever")))
     result.isSuccess must beTrue
   }
 
-  "should fail resolving relative references with custom protocols if option not set(#65)" in {
+  "should fail resolving relative references without any relative URL handler available (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
+      .addUrlResolver(ClasspathUrlProtocolHandler())
```

Comment:
Actually, there is no difference between the classpath relative handler and the classpath protocol handler, except that the protocol handler expects the protocol to be stated explicitly. This is because the classpath relative handler would specify its custom `classpath` protocol only in order to create a valid URL (and we are not using the protocol anywhere else), but since the introduction of the protocol-less variant, this isn't needed anymore because we can use the same custom protocol as we do for the protocol-less variant.
So my understanding of our discussion so far is that protocol handlers handle custom references that are explicitly annotated with the protocol and either are relative or absolute (it's up to the implementer of the `URLStreamHandler` to treat both scenarios) while relative URL handlers handle references that are protocol-less and hence must be relative. 

Regarding your last comment about allowing an `URLStreamHandler` as the single argument to `addUrlProtocolHandler`: the first step would be to extract the default behaviour into a URL resolver as well, but I'm not sure how much work this will be. I'll track that as a separate issue.